src/NuGet.Core/NuGet.Commands/RestoreCommand/RequestFactory/RestoreArgs.cs

```diff
@@ -37,6 +37,15 @@ public class RestoreArgs
 
         public ILogger Log { get; set; }
 
+        /// <summary>
+        /// Sources to use for restore. Overrides Sources
+        /// </summary>
+        public List<SourceRepository> SourceRepositories { get; set; } = new List<SourceRepository>();
```

Comment:
It is an optimization, in VS the SourceRepositories are already cached and data such as v2 vs v3 local folder type, and index.json for v3 are already loaded in memory.

It is a little odd to have both, but the goal of this class is to take all inputs exactly as they are and convert them into the correct restore inputs in the same way. Having nuget.exe, msbuild, and dotnet create SourceRepositories on their own could lead to different results, or would need another helper to collect inputs to assign to this class which is there to collect inputs.